assignments/closure.js

```diff
@@ -8,14 +21,37 @@
 // ==== Challenge 2: Create a counter function ====
 const counter = () => {
   // Return a function that when invoked increments and returns a counter variable.
+  let counter = 0;
+  const countUp = () => counter += 1;
+
+  return countUp;
 };
+
 // Example usage: const newCounter = counter();
 // newCounter(); // 1
 // newCounter(); // 2
+const newCounter = counter();
+console.log(newCounter()); // 1
+console.log(newCounter()); // 2
 
 // ==== Challenge 3: Create a counter function with an object that can increment and decrement ====
 const counterFactory = () => {
   // Return an object that has two methods called `increment` and `decrement`.
   // `increment` should increment a counter variable in closure scope and return it.
   // `decrement` should decrement the counter variable and return it.
+  let counter = 0;
+
+  const counterObject = {
+    increment: () => counter += 1,
```

Comment:
You don't need to use () => counter here as you are then calling increment()

You can do counter +=1 directly and even better counter++ :)